data_structures/src/serializers.rs

```diff
@@ -0,0 +1,572 @@
+        // Create flatbuffer to encode a witnet message
+        match &self.kind {
+            Command::GetPeers => create_get_peers_flatbuffer(
+                &mut builder,
+                GetPeersFlatbufferArgs { magic: self.magic },
+            ),
+            Command::Peers { peers } => create_peers_flatbuffer(
+                &mut builder,
+                PeersFlatbufferArgs {
+                    magic: self.magic,
+                    peers,
+                },
+            ),
+            Command::Ping { nonce } => create_ping_flatbuffer(
+                &mut builder,
+                PingFlatbufferArgs {
+                    magic: self.magic,
+                    nonce: *nonce,
+                },
+            ),
+            Command::Pong { nonce } => create_pong_flatbuffer(
+                &mut builder,
+                &PongFlatbufferArgs {
+                    magic: self.magic,
+                    nonce: *nonce,
+                },
+            ),
+            Command::Verack => {
+                create_verack_flatbuffer(&mut builder, VerackFlatbufferArgs { magic: self.magic })
+            }
+            Command::Version {
+                version,
+                timestamp,
+                capabilities,
+                sender_address,
+                receiver_address,
+                user_agent,
+                last_epoch,
+                genesis,
+                nonce,
+            } => create_version_flatbuffer(
+                &mut builder,
+                &VersionFlatbufferArgs {
+                    magic: self.magic,
+                    version: *version,
+                    timestamp: *timestamp,
+                    capabilities: *capabilities,
+                    sender_address,
+                    receiver_address,
+                    user_agent,
+                    last_epoch: *last_epoch,
+                    genesis: *genesis,
+                    nonce: *nonce,
+                },
+            ),
+        }
+    }
+}
+
+// Encode a flatbuffer from a flatbuffers message
+fn build_flatbuffer(
+    builder: &mut FlatBufferBuilder,
+    message: flatbuffers::WIPOffset<FlatBuffersMessage>,
+) -> Vec<u8> {
+    builder.finish(message, None);
+    builder.finished_data().to_vec()
+}
+
+// Create witnet ip address
+fn create_address(address: FlatBufferAddress) -> Option<Address> {
+    match address.ip_type() {
+        FlatBuffersIpAddress::Ipv4 => address
+            .ip_as_ipv_4()
+            .map(|ipv4| create_ipv4_address(ipv4.ip(), address.port())),
+        FlatBuffersIpAddress::Ipv6 => match address.ip_as_ipv_6() {
+            Some(hextets) => Some(create_ipv6_address(
+                hextets.ip0(),
+                hextets.ip1(),
+                hextets.ip2(),
+                hextets.ip3(),
+                address.port(),
+            )),
+            None => None,
+        },
+        FlatBuffersIpAddress::NONE => panic!(),
```

Comment:
I would mark this with a TODO and create an issue to rethink about the flatbuffers schema and the fields that have to be required

Reply:
I'm now realizing that the function returns an `Option`, sorry. Perhaps change it to return a `None` in that case for the time being? (as well as marking it with the TODO).